src/zenml/zen_stores/secrets_stores/hashicorp_secrets_store.py

```diff
@@ -0,0 +1,634 @@
+        vault_secret_id = self._get_vault_secret_id(secret_id)
+
+        metadata = self._get_secret_metadata_for_secret(
+            secret, secret_id=secret_id
+        )
+
+        created = datetime.utcnow()
+        try:
+            self.client.secrets.kv.v2.create_or_update_secret(
+                path=vault_secret_id,
+                # Store the ZenML secret metadata alongside the secret values
+                secret={
+                    ZENML_VAULT_SECRET_VALUES_KEY: secret.secret_values,
+                    ZENML_VAULT_SECRET_METADATA_KEY: metadata,
+                    ZENML_VAULT_SECRET_CREATED_KEY: created.isoformat(),
+                    ZENML_VAULT_SECRET_UPDATED_KEY: created.isoformat(),
+                },
+                # Do not allow overwriting an existing secret
+                cas=0,
+            )
+        except VaultError as e:
+            raise RuntimeError(f"Error creating secret: {e}")
+
+        logger.debug("Created HashiCorp Vault secret: %s", vault_secret_id)
+
+        secret_model = SecretResponseModel(
+            id=secret_id,
+            name=secret.name,
+            scope=secret.scope,
+            workspace=workspace,
+            user=user,
+            values=secret.secret_values,
+            created=created,
+            updated=created,
+        )
+
+        return secret_model
+
+    def get_secret(self, secret_id: UUID) -> SecretResponseModel:
+        """Get a secret by ID.
+
+        Args:
+            secret_id: The ID of the secret to fetch.
+
+        Returns:
+            The secret.
+
+        Raises:
+            KeyError: If the secret does not exist.
+            RuntimeError: If the HashiCorp Vault API returns an unexpected
+                error.
+        """
+        vault_secret_id = self._get_vault_secret_id(secret_id)
+
+        try:
+            vault_secret = (
+                self.client.secrets.kv.v2.read_secret(
+                    path=vault_secret_id,
+                )
+                .get("data", {})
+                .get("data", {})
+            )
+        except InvalidPath:
+            raise KeyError(f"Secret with ID {secret_id} not found")
+        except VaultError as e:
+            raise RuntimeError(
+                f"Error fetching secret with ID {secret_id} {e}"
+            )
+
+        # The _convert_vault_secret method raises a KeyError if the
+        # secret is tied to a workspace or user that no longer exists. Here we
+        # simply pass the exception up the stack, as if the secret was not found
+        # in the first place, knowing that it will be cascade-deleted soon.
+        return self._convert_vault_secret(
+            vault_secret,
+        )
+
+    def list_secrets(
+        self, secret_filter_model: SecretFilterModel
+    ) -> Page[SecretResponseModel]:
+        """List all secrets matching the given filter criteria.
+
+        Note that returned secrets do not include any secret values. To fetch
```

Comment:
As far as I can see, the secrets actually do contain the values right?

Reply:
No, they dont. The implementation removes them explicilty:

```
                # Remove the secret values from the response
                secret_model.values = {}
                results.append(secret_model)
```